client/src/components/ViewContact.jsx

```diff
@@ -86,36 +66,30 @@ const ViewContact = ({ selectedContactId, setContacts }) => {
         }
       );
 
+      if (!response.ok) {
+        throw new Error("Failed to edit contact");
+      }
+
       await response.json();
 
       const refreshedListData = await fetchContacts();
       setContacts(refreshedListData);
-
-      if (!response.ok) {
-        throw new Error("Failed to edit contact");
-      }
     } catch (error) {
       console.error("Error updating entry: ", error);
+
       setSnackbarMessage("Error editing contact. Please try again.");
-      setContactData((previousData) => ({
-        ...previousData,
-        newName: previousData.originalData.name,
-        newEmail: previousData.originalData.email,
-        newPhone: previousData.originalData.phone,
-        newNotes: previousData.originalData.notes,
-      }));
-      setOpenSnackbar(true);
     } finally {
       setEditButtonClicked(false);
     }
   };
 
   const cancelEdit = () => {
+    setContactData(originalContactDetails.current);
     setEditButtonClicked(false);
   };
 
   const handleSnackbarClose = () => {
-    setOpenSnackbar(false);
+    setSnackbarMessage("");
```

Comment:
Notice how I deleted the `openSnackbar` state completely and can rely on the `snackbarMessage` string to not only hold the text to be shown but to also determine if the snackbar should be closed/opened.